### DISCUSSION TITLE
API documentation

### DISCUSSION THREAD

**Discussion by Abbode** [v2fly/v2ray-core#2839] [General]
hello,  
Everything is working fine for me, except for the API.  
i couldn't find a documentation/guide that is complete.  

If I know how to use it I'd make a good documentation, but I don't...  
Can anyone who uses the API make us a reliable guide...  

Thanks.

**Comment by AGIinspired** [comment]
cant find good docs either but figured out. example: 

in config do:

(top level) 
  "api": {
    "tag": "api",
    "services": [
      "HandlerService",
      "LoggerService",
      "StatsService"
    ]
  },
  "stats": {},

(inbounds)
  {
      "tag": "api-in",
      "listen": "127.0.0.1",  // 0.0.0.0 for over internet
      "port": 10085,
      "protocol": "dokodemo-door",
      "settings": {
        "address": "127.0.0.1"
      }
   },
   
(routing -> rules)
 
     {
        "inboundTag": ["api-in"],
        "outboundTag": "api",
        "type": "field"
      },


then you can do stuff like: 

create a file like new_inbound.json:

{
  "inbounds": [
    {
      "tag": "vmess-in-new",
      "port": 31356,
      "protocol": "vmess",
      "settings": {
        "clients": [
          {
            "id": "UUID",
            "email": "newclient@example.com"
          }
        ]
      }
    }
  ]
}

then do v2ray api adi --server=127.0.0.1:10085 new_inbound.json

Which will add a new inbound to memory, becomes immediately available

**Comment by swuecho** [comment]
https://www.v2ray.com/en/
